### ISSUE TITLE
serveStatic should apply Last-Modified header

### ISSUE THREAD

**Issue by t-mart** [honojs/node-server#189]
Clients can make requests with an [`If-Modified-Since`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since) header to indicate that they only want a data-ful response (200) if the file has changed since the date value in that header. If the file has not changed since that time, the server can respond with a 304 and empty body.

Clients set the value of the `If-Modified-Since` from an earlier response's `Last-Modified` header, which Hono would need to provide.

[`src/serve-static.ts`](https://github.com/honojs/node-server/blob/main/src/serve-static.ts):`serveStatic` doesn't seem to implement this, but seems very close to being able to. We already pull the file's stats for various size headers and the `Date` header.

We'd just have to add another header for `Last-Modified`. Something perhaps like:

```typescript
c.header('Last-Modified', stats.mTime.toUTCString())
```

To maintain backwards-compatibility, a new optional property could be added to the `ServeStaticOptions` object, which is the parameter to `serveStatic`, to gate the adding of this header. For now, I guess it could be a boolean, named something like `setLastModified` (I'm not sure what would be a good name here).

**Comment by yusukebe** [comment]
Hi @t-mart 

Quick response. I think we can implement it as a middleware like [the ETag Middleware](https://hono.dev/docs/middleware/builtin/etag).

**Comment by t-mart** [comment]
@yusukebe, that would make it the most "pluggable", yes.

However, with a general request/response _(not a file mapping serveStatic)_, I'm not sure how the server would (statelessly) know how to set the `Last-Modified` header correctly.

**Comment by yusukebe** [comment]
@t-mart 

> However, with a general request/response _(not a filesystem mapping like serveStatic)_, I'm not sure how the server would (statelessly) know how to set the `Last-Modified` header correctly.

Indeed. As you mentioned, adding this feature to `serveStatic` might be better. And it's a little bit of code.
